File: Backend/sms-backend/src/schemas/academics/academic_grade.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict

from src.schemas.base.base import TimestampSchema, TenantSchema


class AcademicGradeBase(BaseModel):
    """Base schema for AcademicGrade model."""
    name: str
    description: Optional[str] = None
    is_active: bool = True
    sequence: int
    age_range: Optional[str] = None


class AcademicGradeCreate(AcademicGradeBase):
    """Schema for creating a new academic grade."""
    pass


class AcademicGradeUpdate(BaseModel):
    """Schema for updating an academic grade."""
    name: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    sequence: Optional[int] = None
    age_range: Optional[str] = None


class AcademicGradeInDB(AcademicGradeBase, TenantSchema):
    """Schema for AcademicGrade model in database."""
    model_config = ConfigDict(from_attributes=True)


class AcademicGrade(AcademicGradeInDB):
    """Schema for AcademicGrade model response."""
    pass

File: Frontend/sms-client/src/types/academic-grade.ts
export interface AcademicGradeBase {
  name: string;
  description?: string;
  is_active: boolean;
  sequence: number;
  age_range?: string;
}

export type AcademicGradeCreate = AcademicGradeBase;

export interface AcademicGradeUpdate {
  name?: string;
  description?: string;
  is_active?: boolean;
  sequence?: number;
  age_range?: string;
}

export interface AcademicGrade extends AcademicGradeBase {
  id: string;
  tenant_id: string;
  created_at: string;
  updated_at: string;
}

export interface AcademicGradeResponse {
  grades: AcademicGrade[];
  total: number;
  page: number;
  size: number;
  pages: number;
}
